test/globalize_test.rb

```diff
@@ -217,4 +217,28 @@ class GlobalizeTest < MiniTest::Spec
       end
     end
   end
+
+  describe 'with locale' do
+    def perform_with_locale(locale)
+      Thread.new do
+        Globalize.with_locale(locale) do
+          sleep 1
```

Comment:
Just copy paste from https://github.com/globalize/globalize/issues/390#issue-52729253. Feel free to make red test (revert my patch - use ivars as storage) and then make it green with `RequestStore`.